File: hw/sanchez/u1/hw04Arrays/ChickenFarmSimulator2js/ec.edu.espe.chickenfarm.view/ChickenFarmSimulator.js
/**
 * @author Joel Sanchez, The Softwarriors, @ESPE
 */

import readline from 'readline';
import Chicken from '../ec.edu.espe.chickenfarm.model/Chicken.js';

class ChickenFarmSimulator {
    constructor() {
        this.chickens = [];
        this.maxChickens = 100;
        this.rl = readline.createInterface({
            input: process.stdin,
            output: process.stdout
        });
    }

    askQuestion(query) {
        return new Promise(resolve => this.rl.question(query, resolve));
    }

    async run() {
        let counter = 0;
            
        console.log("\nCHICKEN FARM SIMULATOR");

        while (counter < this.maxChickens) {
            console.log(`\nEnter Chicken ${counter + 1}`);
            
            let id = parseInt(await this.askQuestion("Enter id: "));
            let name = await this.askQuestion("Enter name: ");
            let color = await this.askQuestion("Enter color: ");
            let year = parseInt(await this.askQuestion("Enter year: "));
            let month = parseInt(await this.askQuestion("Enter month (0-11): "));
            let day = parseInt(await this.askQuestion("Enter day: "));
            let bornOnDate = new Date(year, month, day);
            let age = parseInt(await this.askQuestion("Enter age: "));
            let isMoltingInput = await this.askQuestion("Is molting? (true/false): ");
            let isMolting = isMoltingInput.toLowerCase() === 'true';

            const chicken = new Chicken(id, name, color, bornOnDate, age, isMolting);
            this.chickens.push(chicken);
            counter++;

            let response = await this.askQuestion("\nDo you want to enter another chicken? (y/n): ");
            if (response.toLowerCase() !== 'y') {
                break;
            }
        }

        console.log("\n List of Registered Chickens\n");
        for (let i = 0; i < counter; i++) {
            console.log(`name --> ${this.chickens[i].getName()}`);
            console.log(`chicken [${i + 1}]${this.chickens[i].toString()}`);
        }

        console.log(`\nTotal registered chickens: ${counter}`);
        this.rl.close();
    }
}

export default ChickenFarmSimulator;

File: my-fresh-copy/ws/sanchez/u1/ws07Arrays/ws07ChickenCoops/src/ec/edu/espe/chickencoops/view/ChickenFarmSimulator.java
package ec.edu.espe.chickencoops.view;

import ec.edu.espe.chickencoops.model.Chicken;
import java.util.Date;


/**
 *
 * @author Joel Sanchez, The Softwarriors, @ESPE
 */

public class ChickenFarmSimulator {
    public static void main(String[] args) {
        
        int id;
        String name;
        String color;
        Date bornOnDate;
        int age;
        boolean isMolting;
        
        id = 1;
        name = "Lucy";
        color = "Brown and white";
        bornOnDate = new Date();
        age = 0;
        isMolting = false;
        
        Chicken chicken;
        Chicken chickens[];
        
        chickens = new Chicken[5];
        
        chickens[0] = new Chicken(id, name, color, bornOnDate, age, isMolting);
        
        id = 2;
        name = "Joel Sanchez";
        color = "Red";
        bornOnDate = new Date(106, 06, 04);
        age = 0;
        isMolting = false;
        
        chickens[1] = new Chicken(id, name, color, bornOnDate, age, isMolting);
        
        id = 3;
        name = "Maria";
        color = "Black";
        bornOnDate = new Date(93, 07, 23);
        age = 0;
        isMolting = false;
        
        chickens[2] = new Chicken(id, name, color, bornOnDate, age, isMolting);
        
        id = 4;
        name = "Juan";
        color = "Blue";
        bornOnDate = new Date(80, 07, 23);
        age = 0;
        isMolting = false;
        
        chickens[3] = new Chicken(id, name, color, bornOnDate, age, isMolting);
        
        id = 5;
        name = "Carlos";
        color = "yellow";
        bornOnDate = new Date(78, 07, 23);
        age = 0;
        isMolting = false;
        
        chickens[4] = new Chicken(id, name, color, bornOnDate, age, isMolting);
        
        System.out.println("QUIZ 2026-04-26 Joel Sanchez");
        
        for(int i = 0; i < 5; i++){
            chickens[i].setId(i + 1);
            System.out.println("name --> " + chickens[i].getName());
            System.out.println("chicken [" + (i + 1) + "]" + chickens[i]);
        } 
        
    }
}
